mypy_django_plugin/transformers/managers.py

```diff
@@ -0,0 +1,153 @@
+from typing import Tuple, List
+
+from mypy.nodes import (
+    GDEF, Argument, FuncDef, MemberExpr, RefExpr, NameExpr, PlaceholderNode,
+    StrExpr, SymbolTableNode, TypeInfo, Var,
+)
+from mypy.plugin import ClassDefContext
+from mypy.types import AnyType, CallableType, Instance, TypeOfAny
+
+from mypy_django_plugin.lib import fullnames, helpers
+from mypy.plugins.common import add_method_to_class
+
+from mypy.types import Type as MypyType
+
+
+def build_unannotated_method_args(method_node: FuncDef) -> Tuple[List[Argument], MypyType]:
+    prepared_arguments = []
+    for argument in method_node.arguments[1:]:
+        argument.type_annotation = AnyType(TypeOfAny.unannotated)
+        prepared_arguments.append(argument)
+    return_type = AnyType(TypeOfAny.unannotated)
+    return prepared_arguments, return_type
+
+
+class ManagerCallback(helpers.DynamicClassFromMethodCallback):
+
+    def copy_method_to_another_class(
+            self,
+            ctx: ClassDefContext,
+            self_type: Instance,
+            new_method_name: str,
+            method_node: FuncDef) -> None:
+        if method_node.type is None:
+            if self.defer_till_next_iteration():
+                return
```

Comment:
Should it fail with an exception instead of `return`? Some error description would be useful for later debugging too. 

In the refactoring, I preferred to fail with exception in all places where there's not enough info to resolve types to be able to do a thing. Usually, it's an implementation bug, not the actual missing info. It's also could be a missing piece of info (no class available in the source code at all), then it should be an error to the user before return. Idea was that if those errors won't be silenced, we will catch bugs sooner, and debugging is easier. 
Those exceptions could be silenced later at the top-level. Or they could be raised with a method, like `raise_incomplete_defn_exception` with a type of exception passed, and internally method introspects smth like `DJANGO_STUBS_DEBUG` environment variable. 

So, please, add raising an exception to all the places where there's not enough info to proceed. 

What do you think about this approach?

Reply:
I think that's just right how things should be done, I will follow this approach